openshift-auth-proxy.js

```diff
@@ -86,14 +107,36 @@ var argv = require('yargs')
   .check(function(args) {
     // yargs#demand doesn't work if we also provide a default, so check for missing/invalid args here
     var errors = [];
-    ["backend", "proxy-cert", "proxy-key", "proxy-tlsopts-file", "session-secret",
-     "client-id", "client-secret", "master-url", "public-master-url", "master-ca",
-     "mutual-tls-ca", "callback-url"].forEach( function(val) {
+    ["backend", "server-cert", "server-key", "server-tlsopts-file", "mutual-tls-ca"].forEach( function(val) {
       if(args[val] == null || args[val].length == 0) {
         errors.push("No value specified for parameter " + val);
       }
     });
+    if (args['auth-mode'] === 'oauth2') {
+      ["oauth-id", "oauth-secret", "master-url", "public-master-url", 'master-ca', "callback-url"].forEach( function(val) {
+        if(args[val] == null || args[val].length == 0) {
```

Comment:
This is more style then anything, but we should settle on using single or double quotes.  We should use '===' instead of '==' as much as possible.